nopol/src/test/java/fr/inria/lille/repair/nopol/NopolTest.java

```diff
@@ -1,18 +1,29 @@
 package fr.inria.lille.repair.nopol;
 
+import fr.inria.lille.commons.synthesis.smt.solver.SolverFactory;
+import fr.inria.lille.repair.ProjectReference;
 import fr.inria.lille.repair.TestUtility;
 import fr.inria.lille.repair.common.config.Config;
 import fr.inria.lille.repair.common.patch.Patch;
 import fr.inria.lille.repair.common.synth.StatementType;
 import org.junit.Ignore;
 import org.junit.Test;
+import xxl.java.junit.TestCase;
+import xxl.java.junit.TestCasesListener;
+import xxl.java.junit.TestSuiteExecution;
 
+import java.net.URLClassLoader;
 import java.util.Collection;
+import java.util.List;
 
 import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class NopolTest extends TestUtility {
```

Comment:
replace inheritance by composition.
split test into setup and asserts.
assert without fields.